P4Plugin/Source/P4Task.cpp

```diff
@@ -444,11 +444,30 @@ bool P4Task::IsLoggedIn()
 		EnableUTF8Mode();
 		loggedIn = p4c->Run(*this, args);
 	}
+	else if (HasServerFingerPrintError(p4c->GetStatus()))
+	{
+		m_Connection->InfoLine("Prompting user for server fingerprint trust");
+		args.resize(1);
+		args.push_back("trust");
+		loggedIn = p4c->Run(*this, args);
+	}
 
 	SendToConnection(*m_Connection, p4c->GetStatus(), MAProtocol);
 	return loggedIn;
 }
 
+static std::string FormatFingerprintMessage(const std::string& statusMessage)
+{
+	std::string noNewlines = statusMessage;
+	for(int i = 0; i < noNewlines.size(); i++)
+	{
+		if(noNewlines[i] == '\n')
+			noNewlines[i] = ' ';
+	}
+	return noNewlines.substr(0, noNewlines.find("To allow connection use the")) +
```

Comment:
What will happen when string won't contain `To allow connection use the` substring? Maybe that does not happen today, but who knows :/

Reply:
It will try to log in without showing the new SSL dialog and should fail as it used to. Should I try to check for another part of the message e.g. "this may be your first attempt to connect to this P4PORT" to determine if it's the same error, or do you suggest testing for the error in some other way?

Reply:
The find should always return >= 0 in d6f7c0f. It also adds some minor unrelated plugin fixes.